changelog/action.yml

```diff
@@ -65,13 +65,29 @@ runs:
         t=$(mktemp)
         printf "# Changelog\n\n" | cat - CHANGELOG.md > "$t" && mv "$t" CHANGELOG.md
 
-        git stash || true
-        git checkout master || true
-        git stash pop || true
-        git add CHANGELOG.md
-        git commit -m "autogen(docs): regenerate and update changelog" || true
-        git pull origin master --rebase || true
-        if [[ "$GITHUB_REF_TYPE" == "branch" ]]; then
-          git push origin "$GITHUB_REF_NAME" || true
+
+        if [[ "$GITHUB_REF_TYPE" != "tag" ]]; then
+          if [[ "$(git log --format=%B -n1)" =~ "autogen"* ]]; then
+            echo "No-op because the previous commit was autogen"
+          else
+            git commit -m "autogen(docs): regenerate and update changelog" || true
+            git pull -ff || true
+            git push origin HEAD:"$GITHUB_REF_NAME" || true
+          fi
+        else
+          if [[ "$(git log --format=%B -n1)" =~ "autogen"* ]]; then
```

Comment:
For tagged commits we do want to run on autogen because it can happen that a release tag coincides with an autogen prefixed commit :)